### PR TITLE
feat(platform)!: withdrawal automatic retries after core rejection

### PR DESCRIPTION
**Pull request by QuantumExplorer**
## Issue being fixed or feature implemented
This PR addresses the handling of expired and broadcasted withdrawal transactions, focusing on rebroadcasting expired transactions and managing broadcasted transactions.

## What was done?
Introduced rebroadcast_expired_withdrawal_documents
Added operations for efficiently managing transaction state changes and moving transactions between queues (e.g., rebroadcast, queue for resigning, or deleting completed transactions).

## How Has This Been Tested?
Added to strategy tests.

## Breaking Changes
This is a breaking change that will be marked as active in v1.4

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [X] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced new fields for managing withdrawal transactions, enhancing transaction handling and configuration limits.
	- Updated core expiration blocks to improve transaction management efficiency.

- **Bug Fixes**
	- Improved logic for handling withdrawal statuses, including filtering chainlocked transactions.

- **Tests**
	- Enhanced withdrawal transaction tests to verify state transitions and broadcasting of expired transactions.

- **Chores**
	- Updated test setup functions to include optional parameters for better initial state configuration.

### CODE REVIEW

File: packages/rs-drive-abci/src/execution/platform_events/withdrawals/rebroadcast_expired_withdrawal_documents/v0/mod.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        if position_of_current_quorum != 0 {
+            tracing::debug!(
+                "Current quorum is not most recent, it is in position {}, not making withdrawals",
+                position_of_current_quorum
+            );
+            return Ok(());
+        }
+        // Version 1 changes on Version 0, by not having the Core 2 Quorum limit.
+        // Hence we can just use the v1 here after the extra logic of v0
+        self.rebroadcast_expired_withdrawal_documents_v1(block_info, transaction, platform_version)
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/platform_events/withdrawals/rebroadcast_expired_withdrawal_documents/v0/mod.rs:37]: No, this is fine.

File: packages/rs-drive/src/util/test_helpers/setup.rs [hunk 2]
```diff
@@ -46,13 +46,15 @@ pub fn setup_drive(drive_config: Option<DriveConfig>) -> Drive {
...
     let drive = setup_drive(Some(DriveConfig {
         batching_consistency_verification: true,
         ..Default::default()
     }));
 
-    let platform_version = PlatformVersion::latest();
+    let platform_version = specific_platform_version.unwrap_or(PlatformVersion::latest());
     drive
         .create_initial_state_structure(None, platform_version)
         .expect("should create root tree successfully");
```
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive/src/util/test_helpers/setup.rs:60]: It's fine.

**Comment by QuantumExplorer** [comment]
Going to merge this in, @shumkov please review when you can.

File: packages/rs-drive/src/util/grove_operations/mod.rs [hunk 3]
```diff
@@ -189,6 +192,29 @@ pub enum BatchDeleteApplyType {
...
+        estimated_key_size: u32,
+        /// What is the estimated value size
+        estimated_value_size: u32,
+        /// The flags length
+        flags_len: FlagsLen,
+    },
+    /// Stateful batch move
+    StatefulBatchMove {
+        /// Are we known to be in a subtree and does this subtree have sums
+        is_known_to_be_subtree_with_sum: Option<(IsSubTree, IsSumSubTree)>,
```
Comment (shumkov) [inline packages/rs-drive/src/util/grove_operations/mod.rs:214]: use struct

**Pull request merged**